mongodb_migrations/cli.py

```diff
@@ -13,65 +13,175 @@
 #     create a file in custom folder
 # upgrade:
 #     run migration files from custom folder
+
+
 class MigrationManager(object):
     config = None
     db = None
     migrations = {}
+    migrations_to_apply = []
     database_migration_names = None
 
     def __init__(self):
         self.config = Configuration()
 
     def run(self):
-        self.db = self._get_mongo_database(self.config.mongo_host, self.config.mongo_port, self.config.mongo_database)
-        files = os.listdir(self.config.mongo_migrations_path)
+        # Connect to the DB
+        self.db = self._get_mongo_database(
+            self.config.mongo_host, self.config.mongo_port,
+            self.config.mongo_database)
+
+        # Discover the migration files
+        self._discover_migration_files(self.config.mongo_migrations_path)
+
+        migrations_in_order = self._get_migrations_in_order()
+
+        print("Migrations found: %s" % " <-> ".join(migrations_in_order))
+
+        current_version = self._get_current_version()
+        if current_version:
+            last_migration_idx = migrations_in_order.index(
+                current_version)
+        else:
+            last_migration_idx = -1
+        if self.config.execution == Execution.UPGRADE:
+            self.migrations_to_apply = \
+                migrations_in_order[last_migration_idx+1:]
+        else:
+            # Downgrade is applied by one step
+            try:
+                self.migrations_to_apply = \
+                    migrations_in_order[
+                        max(0, last_migration_idx-2):last_migration_idx+1]
+                self.migrations_to_apply.reverse()
+            except IndexError:
+                self.migrations_to_apply = None
+
+        self._migrate()
+        sys.exit(0)
+
+    def _discover_migration_files(self, migrations_path):
+        files = os.listdir(migrations_path)
         for file in files:
-            result = re.match('^(\d+)[_a-z]*\.py$', file)
+            result = re.match('^([_a-zA-Z0-9]+)_[_a-zA-Z0-9]*\.py$', file)
```

Comment:
This will ask for an underscore symbol in each file and will allow to use alfanumeric character for the label and description 
> (<label>_<description>.py)